### DISCUSSION TITLE
Machine Learning

### DISCUSSION THREAD

**Discussion by mssanamateen** [mssanamateen/MLAlgorithms#9] [Q&A] [answered]
What is training and testing in Machine Learning?

**Comment by BelureSanket** [comment, marked as answer]
Training involves teaching the model using a dataset, while testing evaluates its performance on unseen data to ensure it can make accurate predictions beyond the training data.

**Comment by MohammedAbrarAli** [comment]
Training data is used to train the machine-learning model. The more training data the model has, the better it can make predictions. The data that is used to evaluate the model and check its performance is known as the testing data.

**Comment by Anashussain17** [comment]
Training is the process of teaching a model to make predictions by presenting it with labeled examples from which it can learn. 
Testing is the process of evaluating the performance of a trained model on a dataset that it has not seen during training

**Comment by Mazin7d4** [comment]
Training is a process of using the training data and a set of training algorithms to develop a model which can make predictions.

Testing is the process of using the test data and the model developed during training to check the performance and accuracy of the model. In testing, it can be verified whether the predictions by the trained model are accurate enough in recognizing the unknown data.

**Comment by lavachary** [comment]
Training in ML refers to the process of teaching a model to recognize patterns and make predictions or decisions based on input data.
whereas testing refers to evaluation of a trained model's performance on unseen data to assess its ability to generalize and make accurate predictions or decisions.
